rosbag2_compression/test/rosbag2_compression/test_zstd_compressor.cpp

```diff
@@ -76,13 +102,32 @@ class CompressionHelperFixture : public rosbag2_test_common::TemporaryDirectoryF
 
   void SetUp() override
   {
+    allocator_ = rcutils_get_default_allocator();
+    message_ = create_garbage_string();
+
     rclcpp::init(0, nullptr);
   }
 
   void TearDown() override
   {
     rclcpp::shutdown();
   }
+
+  std::string deserialize_message(std::shared_ptr<rcutils_uint8_array_t> serialized_message)
+  {
+    std::unique_ptr<uint8_t[]> copied(new uint8_t[serialized_message->buffer_length + 1]);
+    std::copy(
+      serialized_message->buffer,
+      serialized_message->buffer + serialized_message->buffer_length,
+      copied.get());
+    copied.get()[serialized_message->buffer_length] = '\0';
+    std::string message_content(reinterpret_cast<char *>(copied.get()));
+    return message_content;
+  }
+
+  rcutils_allocator_t allocator_;
+  std::string message_;
+  size_t compressed_length_{976};  // manually calculated, could change if compression params change
```

Comment:
I know this is a test, but can this be a constant somewhere?
Is there a reason this can't be programmatically determined?

Reply:
Yeah, it could definitely be pulled out as a constant.  I'm not sure if there's a good way to programmatically determine it; there's not a way (as far as I know, maybe I missed it) to calculate beforehand how large the data will be after it's compressed.  The way I determined this was to compress the data, then decompress it, then compare them to verify they're the same, which is good for an end-to-end test but more steps than you want to do for just a quick test of the compression alone.